include/vara/Feature/FeatureModel.h

```diff
@@ -438,16 +446,17 @@ template <typename... Rules> class FeatureModelConsistencyChecker {
 struct EveryFeatureRequiresParent {
   static bool check(FeatureModel &FM) {
     return std::all_of(FM.begin(), FM.end(), [](Feature *F) {
-      return F->isRoot() || (F->getParent() != nullptr);
+      return llvm::isa<RootFeature>(F) ||
+             (F->getParent() && F->getParentFeature());
```

Comment:
Why do we need the extra `getParentFeature` check?

Reply:
In this loop we only iterate over feature nodes, so relationships parents are not covered. If we assert, that each feature has an ancestor, this implies that also every relationship has a parent of some kind (assuming we prohibit directly nested relationships).

Reply:
But how does checking if the parent is a Feature help if we don't iterate over the other nodes?

Reply:
If we have a feature in a relationship, the parent node will only be of that kind. So if we find an parent feature (which can't be the actual parent), this means, that the relationship node itself has to have a parent, which again should be a feature. So we implicitly check if every relationship has a parent as well.

tl;dr: If we check that features are connected to a parent feature, this may be transitive over a relationship -- so the relationship itself has to have a parent in the first place.